app/src/main/java/org/p2p/wallet/smsinput/striga/StrigaSmsInputInteractor.kt

```diff
@@ -74,22 +88,29 @@ class StrigaSmsInputInteractor(
                 ).toFailureResult()
             }
             else -> {
-                strigaUserRepository.resendSmsForVerifyPhoneNumber()
-                    .onTypedFailure<StrigaDataLayerError.ApiServiceError> {
-                        when (it.errorCode) {
-                            StrigaApiErrorCode.EXCEEDED_VERIFICATION_ATTEMPTS -> {
-                                setExceededVerificationAttempts()
-                            }
-                            StrigaApiErrorCode.EXCEEDED_DAILY_RESEND_SMS_LIMIT -> {
-                                setExceededResendAttempts()
-                            }
-                            else -> Unit
-                        }
-                    }
+                null
             }
         }
     }
 
+    suspend fun resendSms(): StrigaDataLayerResult<Unit> {
+        return getExceededLimitsErrorIfPresent() ?: strigaUserRepository.resendSmsForVerifyPhoneNumber()
```

Comment:
and again, i suggest extracting the whole block from the left of ?: to a method

Reply:
@gslevinkov Do you suggest moving logic from getExceededLimitsErrorIfPresent to strigaUserRepository.resendSmsForVerifyPhoneNumber ? 
This function intentionally was separated because it is also used in the presenter to get this error without an api call. This means we would have to call resendSms even when it's not necessary